pkg/cmd/cli/cmd/startbuild.go

```diff
@@ -102,10 +103,30 @@ func RunStartBuild(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, args
 		return cmdutil.UsageError(cmd, "Must pass a name of a BuildConfig or specify build name with '--from-build' flag")
 	}
 
-	name := buildName
-	isBuild := true
-	if len(name) == 0 {
-		name = args[0]
+	namespace, _, err := f.DefaultNamespace()
+	if err != nil {
+		return err
+	}
+
+	var (
+		name     = buildName
+		resource = ""
+		isBuild  = true
```

Comment:
having both `isBuild` and `resource` is confusing... seems like this should start out as
```
name = buildName
resource = "builds"
```

and the code below that switches on isBuild should switch on resource